pkg/steps/release/snapshot.go

```diff
@@ -37,12 +39,19 @@ func (r *releaseSnapshotStep) Run(ctx context.Context) error {
 }
 
 func (r *releaseSnapshotStep) run(ctx context.Context) error {
-	_, _, err := snapshotStream(ctx, r.client, r.config.Namespace, r.config.Name, r.jobSpec.Namespace, r.name)
+	builtImages := sets.New[string]()
+	if r.config.IncludeBuiltImages {
+		for _, image := range r.images {
+			builtImages.Insert(string(image.To))
+		}
+	}
+	// The images that were built in the test will be tagged into the "stable" image stream by OutputImageTagStep
+	_, _, err := snapshotStream(ctx, r.client, r.config.Namespace, r.config.Name, r.jobSpec.Namespace, r.name, builtImages)
```

Comment:
Before this PR, we import all of the tags including the one that are built in the test and then expect that they are overridden by [OutputImageTagStep](https://docs.ci.openshift.org/docs/internals/steps/#outputimagetagstep).

In the rehearsals, I found out the image points to QCI-APPCI too.
I _believe_ that it is the racing condition on importing from the snapshotting and the tag from OutputImageTagStep.
So here, the imports ignores the built images at snapshotting and wait for OutputImageTagStep to do the job.